src/psyclone/generator.py

```diff
@@ -428,22 +428,44 @@ def main(args):
         psy_str = str(psy)
         alg_str = str(alg)
     if args.oalg is not None:
-        my_file = open(args.oalg, "w")
-        my_file.write(alg_str)
-        my_file.close()
+        write_unicode_file(alg_str, args.oalg)
     else:
         print("Transformed algorithm code:\n%s" % alg_str)
 
     if not psy_str:
         # empty file so do not output anything
         pass
     elif args.opsy is not None:
-        my_file = open(args.opsy, "w")
-        my_file.write(psy_str)
-        my_file.close()
+        write_unicode_file(psy_str, args.opsy)
     else:
         print("Generated psy layer code:\n", psy_str)
 
 
+def write_unicode_file(contents, filename):
+    '''
+    Wrapper routine that ensures that a string is encoded as unicode before
+    writing to file in both Python 2 and 3.
+
+    :param str contents: string to write to file.
+    :param str filename: the name of the file to create.
+
+    '''
+    import six
+    import io
+
+    if six.PY2:
```

Comment:
This is an optional request. I think the following is clearer and more future proof, but I will accept the code as is

```
if six.py2:
    contents = ...
    encoding = ...
elif six.py3:
    encoding = ...
else:
    raise ...
```

Reply:
I was going to disagree but upon reflection decided you were right :-)

Reply:
Thanks.